tests/test_sudoku.py

```diff
@@ -121,3 +124,22 @@ def test_sudoku_build_grid():
     random.seed(10)
     sudoku = Sudoku(Sudoku.build_grid())
     assert sudoku.get_grid() == BUILD_GRID_SOLUTION
+
+
+@pytest.mark.skip(reason="Alternative exclusion bug")
```

Comment:
Причина не "Alternative exclusion bug". Этот баг решён.
На  самом деле мы утопаем в рекурсии потому, что существует слишком большое количество способов решить это задание. При этом, судя по всему, решений не так уж много, но способов их получить да.

Можно было бы принять подход: если встречаем хотя бы два разных способа решить задание -- отказываемся его решать.
Однако опыты показывают, что даже это не спасает. Цифр может быть меньше, при этом достаточно, чтобы погрузиться в рекурсию, но недостаточно чтобы найти хотя бы одно решение за разумное время. 
Наши классические подходы к решению с этим судоку не справятся.

Поэтому тест нужно пока оставить `mark.skip`. А `reason` я бы сформулировал так: 
```suggestion
@pytest.mark.skip(reason="Recursion calls take too much time")
```